test/gtest/ucp/test_ucp_context.cc

```diff
@@ -107,11 +107,13 @@ UCS_TEST_P(test_ucp_version, version_string) {
 
     ucp_get_version(&major_version, &minor_version, &release_number);
 
-    char buffer[256];
-    snprintf(buffer, sizeof(buffer), "%d.%d.%d", major_version, minor_version,
-             release_number);
+    std::string string_version     = std::to_string(major_version) + '.' +
+                                     std::to_string(minor_version) + '.' +
+                                     std::to_string(release_number);
+    std::string ucp_string_version = ucp_get_version_string();
```

Comment:
is it fix or just improvement?

Reply:
this is a fix, string version also includes 'extra' version and if it's not empty the strings are not equal

Reply:
i don't see where string version also includes 'extra' version..

Reply:
https://github.com/Mellanox/ucx/blob/009c38c65fd69f212e59624eb264a85e9eabd2eb/src/ucp/core/ucp_version.c.in#L17

Reply:
maybe add extra_version to string_version as well, to keep EXPECT_EQ without using substr?

Reply:
`@VERSION@` includes extra version that's the issue. If we want to have strong equal here, then need to return `major.minor.patch` version in string

Reply:
why can't we also compare with extra version? because not API that returns it?

Reply:
> because not API that returns it?

yes

Reply:
ok